### ISSUE TITLE
Flaky-test: ConsumedLedgersTrimTest.testConsumedLedgersTrimNoSubscriptions

### ISSUE THREAD

**Issue by lhotari** [apache/pulsar#10380]
ConsumedLedgersTrimTest is flaky. The testConsumedLedgersTrimNoSubscriptions test method fails sporadically.

[example failure](https://github.com/apache/pulsar/pull/10371/checks?check_run_id=2434862981#step:9:454)

```
Error:  Tests run: 5, Failures: 1, Errors: 0, Skipped: 3, Time elapsed: 23.508 s <<< FAILURE! - in org.apache.pulsar.broker.service.ConsumedLedgersTrimTest
Error:  testConsumedLedgersTrimNoSubscriptions(org.apache.pulsar.broker.service.ConsumedLedgersTrimTest)  Time elapsed: 5.276 s  <<< FAILURE!
java.lang.AssertionError: expected [3:6:-1:0] but found [-1:-1:-1]
	at org.testng.Assert.fail(Assert.java:99)
	at org.testng.Assert.failNotEquals(Assert.java:1037)
	at org.testng.Assert.assertEqualsImpl(Assert.java:140)
	at org.testng.Assert.assertEquals(Assert.java:122)
	at org.testng.Assert.assertEquals(Assert.java:617)
	at org.apache.pulsar.broker.service.ConsumedLedgersTrimTest.testConsumedLedgersTrimNoSubscriptions(ConsumedLedgersTrimTest.java:153)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:132)
	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:45)
	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:73)
	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:11)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by lhotari** [comment]
Previous issue about the same test: #9409 . A fix was made in #9420 .

@315157973 would you mind taking a look at the remaining flakiness in this test?

**Comment by 315157973** [comment]
I have executed it about 1000 times locally, but it can't be reproduced.

**Comment by lhotari** [comment]
> I have executed it about 1000 times locally, but it can't be reproduced.

Yes I have noticed the same. Some issues reproduce only when you limit CPU resources. CI runs with 2 cores and it's usually effective to run tests within a docker container with limited resources. Docker can limit resources with `--cpus` and `--cpuset-cpus` parameters for `docker run`. I have shared some scripts that I use for development in https://github.com/lhotari/pulsar-contributor-toolbox#ptbx_until_test_fails_in_docker
